content/docs/how-to-guides/sqlx.md

```diff
@@ -52,32 +52,20 @@ In-line Javascript can be used anywhere SQL is written in order to dynamically m
 
 ### Built-in functions
 
-Built in functions have special functionality should be executed within [in-line Javascript](#in-line-javascript).
+Built in functions have special functionality and can be executed either within [in-line Javascript](#in-line-javascript) or [javascript blocks](#javascript-blocks).
 
-#### `ref()`
+For all built in functions, see [`ITableContext` in the API reference](/reference#ITableContext). Some useful examples can be found here:
 
-`ref()` enables you to easily reference another dataset in your project without having to provide the full SQL dataset name. `ref()` also adds the referenced dataset to the set of dependencies for the query.
+- [Reference another dataset in your project with `ref()`](datasets/#referencing-other-datasets).
 
-An example of `ref()` being used to add a dependency is [here](datasets/#referencing-other-datasets).
+- [Retrieve the name of the current dataset with `self()`](incremental-datasets/#a-simple-example).
 
-#### `resolve()`
-
-`resolve()` works similarly to `ref()`, but doesn't add the dataset to the dependency list for the query.
-
-#### `self()`
-
-`self()` returns the name of the current dataset. If the default schema or dataset name is overridden in the `config{}` block, `self()` will return the full and correct dataset name.
-
-An example of `self()` being used to set up incremental tables is [here](incremental-datasets/#a-simple-example).
+- [Execute code only if the script is for an incremental dataset using `incremental()`](incremental-datasets/#conditional-code-if-incremental).
 
 ## Additional Features
 
-- [Pre-operations and post-operations](#pre-operations-and-post-operations): Pre and post operations are only valid for some table types.
-
-### Pre-operations and post-operations
-
-Pre-operation and post-operation blocks are defined in SQLX by writing `pre_operations { }` and `post_operations { }` respectively.
+- **Post-operations**: defined in SQLX by writing `post_operations { }`, SQL written inside will be executed after the central SQL. This can be useful for granting permissions, as can be seen in the [publishing datasets guide](/how-to-guides/datasets/#example-granting-dataset-access-with-post_operations). **Actions may only include pre_operations if they create a dataset**, for example with `type: "table"` or `type: "view"` or `type: "incremental"` in their config.
```

Comment:
this is an overview, IMO we should leave the details of where it's ok to use `pre_operations` etc in the specific guide (`/datasets`)

Reply:
Hm I disagree, I think the information here is more suited in this guide than in `/datasets`, as when publishing a dataset, a dataset will always be created; users searching about how to use post_operations in SQLX when trying to understand why they they get an error while defining it with `inline` would end up here first.

isn't in `/datasets`